src/libraries/System.Net.Quic/src/System/Net/Quic/QuicListener.cs

```diff
@@ -333,7 +333,9 @@ private unsafe int HandleEventNewConnection(ref NEW_CONNECTION_DATA data)
         SslClientHelloInfo clientHello = new SslClientHelloInfo(data.Info->ServerNameLength > 0 ? Marshal.PtrToStringUTF8((IntPtr)data.Info->ServerName, data.Info->ServerNameLength) : "", SslProtocols.Tls13);
 
         // Kicks off the rest of the handshake in the background, the process itself will enqueue the result in the accept queue.
-        StartConnectionHandshake(connection, clientHello);
+        // this also makes sure the connection options callback provided by the user is not invoked
+        // from the MsQuic thread and cannot delay acks or other operations on other connections.
+        _ = Task.Run(() => StartConnectionHandshake(connection, clientHello));
```

Comment:
@stephentoub would calling `await Task.Yield()` at the beginning of `StartConnectionHandshake` achieve the same/similar thing? E.g. offloading the work outside of MsQuic thread.

Reply:
Yes... except you'd want to use `await Task.CompletedTask.ConfigureAwait(ConfigureAwaitOptions.ForceYielding);` if there's any chance the current SynchronizationContext / TaskScheduler is not the default, as Task.Yield's continuation would go back there. You can see examples of Task.Runs like this one being replaced with that in https://github.com/dotnet/runtime/pull/87067.